ios/RNSScreen.mm

```diff
@@ -251,6 +251,13 @@ - (void)setStatusBarHidden:(BOOL)statusBarHidden
   _statusBarHidden = statusBarHidden;
   [RNSScreenWindowTraits assertViewControllerBasedStatusBarAppearenceSet];
   [RNSScreenWindowTraits updateStatusBarAppearance];
+
+  // As the status bar could change its visibility, we need to calculate header
+  // height for the correct value in `onHeaderHeightChange` event when navigation
+  // bar is not visible.
+  if (self.controller.navigationController.navigationBarHidden && !self.isModal) {
+    [self.controller calculateAndNotifyHeaderHeightChangeIsModal:NO];
+  }
```

Comment:
I do not really get why we need to recalculate header height on status bar visibility change? Is it because we use status bar height in case: non-modal view && navigation controller is `nil` or navigation bar is hidden? If so, what happens if the status bar is hidden? Are correct values reported?

Reply:
That's because `viewDidLayoutSubviews` is not being called when the status bar visibility is being changed 😃 
Also values seems to be correct there.